src/routes/LocationPage/index.js

```diff
@@ -40,38 +44,37 @@ class PageAdapter extends React.Component {
         location={this.props.location}
       />
       <Content url={url} hierarchy={hierarchy}/>
-      <PDFButton languageCode={this.props.language} locationCode={this.props.location} page={hierarchy.top()} />
+      <PDFButton languageCode={this.props.language}
+                 locationCode={this.props.location}
+                 page={hierarchy.top()}/>
     </div>
   }
 }
 
+function mapStateToProps (state) {
+  return {
+    language: state.router.params.language,
+    path: state.router.params['_'] // _ contains all the values from *
+  }
+}
+
+const FetchingContentWrapper = compose(
+  connect(mapStateToProps),
+  withFetcher(PAGE_ENDPOINT)
+)(ContentWrapper)
+
 class LocationPage extends React.Component {
   static propTypes = {
-    location: PropTypes.string.isRequired,
-    path: PropTypes.string,
-    language: PropTypes.string.isRequired
+    location: PropTypes.string.isRequired
   }
 
   render () {
     return (
       <RichLayout location={this.props.location}>
-        <PageFetcher>
-          <PageAdapter
-            location={this.props.location}
-            language={this.props.language}
-            path={this.props.path} />
-        </PageFetcher>
+        <FetchingContentWrapper location={this.props.location}/>
       </RichLayout>
     )
   }
 }
 
-function mapStateToProps (state) {
-  return {
-    location: state.router.params.location,
-    language: state.router.params.language,
-    path: state.router.params['_'] // _ contains all the values from *
-  }
-}
-
-export default connect(mapStateToProps)(LocationPage)
+export default connect((state) => ({location: state.router.params.location}))(LocationPage)
```

Comment:
We don't need location in the LocationPage props, if you add the location-prop in the mapStateToProps for the FetchingContentWrapper. Also the RichLayout is connected to the state itself.
So we disconnect the LocationPage from the state 🎉

Reply:
done